### PR TITLE
[Core][Geometry] adding Check

### LINKED ISSUE #583: Condition::Check does not throw an error for colapsed (zero area) triangles
In PR #582 I found out that Condition::Check only verifies that the element size is not negative, but happily accepts zero size geometries. I understand this was done so that point conditions do not fail the check, however this also means that degenerate geometries (e.g. a triangle with the three points on a line) will pass the test too, which is not very useful.

Can we think of a more comprehensive way of checking for validity in Condition::Check? I can think of different alternatives, but none of them is perfect:
- We leave things as they are: we catch inverted geomteries but not degenerate ones.
- We force a check for area <= 0, but make it clear that derived classes are not forced to also call the base class checks (so that point conditions use a completely different check method, implemented by their developer).
- We check for area <= 0, but exclude point conditions from this check (for example, by also checking geometry type and/or number of points). This does not cover all cases (I can think of using zero-length lines to describe contact-type conditions, for example), but should work in most common situations.

Mentioning @KratosMultiphysics/structural-mechanics, since most point conditions are on the structural side.

### PR DESCRIPTION
**Pull request by philbucher**
**Description**
This PR adds a `Check` method to the `Geometry`. This is a todo from long ago (#583).
Also I think it can be very useful for the NURBS stuff of @tteschemacher 

closes #583

### CODE REVIEW

File: kratos/geometries/geometry.h [hunk 1]
```diff
@@ -3461,6 +3461,18 @@ class Geometry
         return rResult;
     }
 
+    virtual int Check() const
+    {
+        KRATOS_TRY
+
+        const double domain_size = DomainSize();
+        KRATOS_ERROR_IF(domain_size <= 0.0) << "Geometry # " << std::to_string(mId) << " has non-positive size " << domain_size << std::endl;
```
Comment (rubenzorrilla) [inline kratos/geometries/geometry.h]: I'm very much in favor of adding the `Check` to the geometry. The point is that the error thrown with the geometry Id might be problematic as these can have the self assigned id, which would difficult debugging.

I think we should still throw the error message in both `Element` and `Condition` as these always have a meaningful id which can be backtraced in the preprocessor.

Comment (philbucher) [reply kratos/geometries/geometry.h]: I see your point but then we don’t need this method if we anyway throw in the element

Comment (philbucher) [reply kratos/geometries/geometry.h]: We can try to issue the element id in the KRATOS_CATCH of the Check of Element

@roigcarlo is this possible?
Or must the input for KRATOS_CATCH be known at compile time?

**Comment by tteschemacher** [comment]
That could be indeed pretty good 🤠 thanks!

It's very common that the input is dodgy, if it doesn't come from our own interface...

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**